src/core/raster/qgsrasterlayer.cpp

```diff
@@ -125,7 +125,7 @@ QgsRasterLayer::QgsRasterLayer( const QString &uri,
   QgsDebugMsgLevel( QStringLiteral( "Entered" ), 4 );
   setProviderType( providerKey );
 
-  QgsDataProvider::ProviderOptions providerOptions { options.transformContext };
+  QgsDataProvider::ProviderOptions providerOptions { options.transformContext, ( mReadFlags & QgsMapLayer::FlagTrustLayerMetadata ) ? true : false };
```

Comment:
This is a fragile way of constructing ProviderOptions -- if someone reorganised the members in the future then this code would break. Better to make an explicit constructor for ProviderOptions so that this is using stable api.

Reply:
I have removed my changes on `struct ProviderOptions` and added `ReadFlags`